spec/support/macros.rb

```diff
@@ -0,0 +1,11 @@
+def set_current_user(user=nil)
+  session[:user_id] = (user || Fabricate(:user)) .id
+end
+
+def sign_in(a_user=nil)
+  user = a_user || Fabricate(:user)
```

Comment:
You should refactor this to use conditional assignment, or when you look at memoization:

```ruby
def sign_in(user=nil)
  user ||= Fabricate(:user)
```